apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -185,6 +198,25 @@ export class UpdateIntegration {
       updatePayload.conditions = command.conditions;
     }
 
+    const isRemoveNovuBrandingDefined = typeof command.removeNovuBranding !== 'undefined';
+    const isRemoveNovuBrandingChanged =
+      isRemoveNovuBrandingDefined && existingIntegration.removeNovuBranding !== command.removeNovuBranding;
+
+    if (isRemoveNovuBrandingChanged) {
+      const isImprovedBillingEnabled = await this.getFeatureFlag.execute(
+        GetFeatureFlagCommand.create({
+          userId: 'system',
+          environmentId: 'system',
+          organizationId: command.organizationId,
+          key: FeatureFlagsKeysEnum.IS_IMPROVED_BILLING_ENABLED,
+        })
+      );
+
+      if (isImprovedBillingEnabled) {
+        updatePayload.removeNovuBranding = await this.setRemoveNovuBranding(command);
+      }
+    }
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** can you please cover this code with e2e tests?

Reply:
Done ✅